fe1-web/src/features/events/components/EventList.tsx

```diff
@@ -1,159 +1,137 @@
 import STRINGS from 'resources/strings';
 
+import { categorizeEventsByTime } from '../functions';
 import { EventHooks } from '../hooks';
 import { EventState } from '../objects';
 import { makeEventListSelector } from '../reducer';
 import EventListItem from './EventListItem';
 
-const categorizeEventsByTime = (time: Timestamp, events: EventState[]) => {
-  const t = time.valueOf();
-
-  const pastEvents: EventState[] = [];
-  const currentEvents: EventState[] = [];
-  const futureEvents: EventState[] = [];
-
-  events.forEach((e: EventState) => {
-    if ((e.end && e.end < t) || (!e.end && e.start < t)) {
-      pastEvents.push(e);
-      return;
-    }
-    if (e.start > t) {
-      futureEvents.push(e);
-      return;
-    }
-    currentEvents.push(e);
-  });
-
-  return [pastEvents, currentEvents, futureEvents];
-};
+type NavigationProps = CompositeScreenProps<
+  StackScreenProps<LaoEventsParamList, typeof STRINGS.navigation_lao_events_home>,
+  CompositeScreenProps<
+    StackScreenProps<LaoParamList, typeof STRINGS.navigation_lao_events>,
+    StackScreenProps<AppParamList, typeof STRINGS.navigation_app_lao>
+  >
+>;
 
 /**
  * Collapsible list of events: list with 3 sections corresponding
  * to 'past', 'present' and 'future' events.
  */
 const EventList = () => {
   const laoId = EventHooks.useAssertCurrentLaoId();
+  const navigation = useNavigation<NavigationProps['navigation']>();
 
   const eventListSelector = useMemo(() => makeEventListSelector(laoId.valueOf()), [laoId]);
   const events = useSelector(eventListSelector);
 
-  const [{ pastEvents, currentEvents, futureEvents }, setEvents] = useState<{
+  const [{ pastEvents, currentEvents, upcomingEvents }, setEvents] = useState<{
     pastEvents: EventState[];
     currentEvents: EventState[];
-    futureEvents: EventState[];
-  }>(() => {
-    const [newPastEvents, newCurrentEvents, newFutureEvents] = categorizeEventsByTime(
-      Timestamp.EpochNow(),
-      events,
-    );
-
-    return {
-      pastEvents: newPastEvents,
-      currentEvents: newCurrentEvents,
-      futureEvents: newFutureEvents,
-    };
-  });
+    upcomingEvents: EventState[];
+  }>(() => categorizeEventsByTime(Timestamp.EpochNow(), events));
 
-  const [showUpcoming, setShowUpcoming] = useState(true);
   const [showCurrent, setShowCurrent] = useState(true);
-  const [showPast, setShowPast] = useState(false);
+  const [showPast, setShowPast] = useState(currentEvents.length === 0);
 
   useEffect(() => {
-    const interval = setInterval(() => {
-      const [newPastEvents, newCurrentEvents, newFutureEvents] = categorizeEventsByTime(
-        Timestamp.EpochNow(),
-        events,
-      );
-
-      setEvents({
-        pastEvents: newPastEvents,
-        currentEvents: newCurrentEvents,
-        futureEvents: newFutureEvents,
-      });
-    }, 1000);
+    const interval = setInterval(
+      () => setEvents(categorizeEventsByTime(Timestamp.EpochNow(), events)),
```

Comment:
Yes that is true but I guess for this we would need to compute the minimum time delta of a categorization change of any event? Or do you see a better approach? I do not think we can leverage redux for this since time passes without the state changing?

Reply:
I don't see a better approach. Indeed, we'd have to compute a delta. As long as we don't have 100+ events, we should be fine 😅 I'm happy to make this a priority when we start deploying (soon, hopefully).